components/comment/index.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+export interface CommentProps {
+  /** List of action items rendered below the comment content */
+  actions?: Array<React.ReactNode>;
+  /** The element to display as the comment author. */
+  author?: string;
+  /** The element to display as the comment avatar - generally an antd Avatar */
+  avatar?: React.ReactNode;
+  /** className of comment */
+  className?: string;
+  /** The main content of the comment */
+  content: React.ReactNode;
+  /** Nested comments should be provided as children of the Comment */
+  children?: any;
+  /** Additional style for the comment content */
+  contentStyle?: React.CSSProperties;
+  /** Additional style for the comment avatar */
+  avatarStyle?: React.CSSProperties;
+  /** Optional ID for the comment */
+  id?: string;
+  /** Additional style for the comment inner wrapper */
+  innerStyle?: React.CSSProperties;
+  /** Comment prefix defaults to '.ant-comment' */
+  prefixCls?: string;
+  /** Additional style for the comment */
+  style?: React.CSSProperties;
+  /** A time element containing the time to be displayed */
+  time?: React.ReactNode;
+  /** Direction of the comment left or right */
+  direction?: 'left' | 'right';
```

Comment:
Should we provide this `direction` props? We may provide one official direction but user can change it though additional css.
Just my thinking, welcome discuss.

Reply:
This part is tricky, some projects use both sides for communication and some don't. For example Airbnb:
<img width="706" alt="screenshot 2018-10-29 at 16 31 01" src="https://user-images.githubusercontent.com/9677898/47660793-0e866200-db98-11e8-8e0d-5fdaaf99c87a.png">
I think without this prop user can't do it himself with css, because dom is different. On the other side without `direction` code is more clean and easy to maintain 😄what do you think?

Reply:
Since flex provide the [order ability](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Ordering_Flex_Items), user can customize.
And the upper sample is more like a chat box.